### PR TITLE
Component play should not execute if the component is not initialized

### PR DESCRIPTION
**Pull request by dmarcos**
**Description:**

There's a bogus situation if `play` is called on a component that has not been yet initialized and the logic relies on something done in the `init` method. 

The bug has manifested when the camera system pauses the inactive camera entity and plays the new active one. With the default camera the camera system calls play on the entity. The camera component is initialized before `look-controls` and its `play` method invoked before `init`. The `play` method of `look-controls` expects the methods to be bound before attaching the event listeners. 

**Changes proposed:**
Ensure that `play` is never invoked before `init` and also prevent play/pause logic to be executed multiple times if the component is already playing or paused respectively.

### CODE REVIEW

File: src/core/component.js [hunk 1]
```diff
@@ -259,10 +260,14 @@ module.exports.registerComponent = function (name, definition) {
   NewComponent = function (el, attr) {
     Component.call(this, el, attr);
   };
+
   NewComponent.prototype = Object.create(Component.prototype, proto);
   NewComponent.prototype.name = name;
   NewComponent.prototype.constructor = NewComponent;
   NewComponent.prototype.system = systems && systems.systems[name];
+  // Wraps method to meet needed gurantees
```
Comment (ngokevin) [inline src/core/component.js]: `Wraps methods to meet needed guarantees.`

Can also explain what the "guarantees" are.

File: src/core/component.js [hunk 2]
```diff
@@ -203,6 +203,7 @@ Component.prototype = {
     if (!this.initialized) {
       this.init();
       this.initialized = true;
+      if (el.isPlaying) { this.play(); }
```
Comment (ngokevin) [inline src/core/component.js]: Sort of confusing since seems like it should read `should be playing` rather than `is playing`. A comment might help.

File: src/core/component.js [hunk 3]
```diff
@@ -346,3 +351,40 @@ function extendProperties (dest, source, isSinglePropSchema) {
   if (isSinglePropSchema) { return source; }
   return utils.extend(dest, source);
 }
+
+/**
+ * Pause component by removing tick behavior and calling pause handler.
+ *
+ * @param component {object} - Component to pause.
```
Comment (ngokevin) [inline src/core/component.js]: Incorrect parameters in docstring.

File: src/core/component.js [hunk 4]
```diff
@@ -346,3 +351,40 @@ function extendProperties (dest, source, isSinglePropSchema) {
...
+ * @param sceneEl {Element} - Scene, needed to remove the tick behavior.
+ */
+function wrapPause (pauseMethod) {
+  return function pause () {
+    var sceneEl = this.el.sceneEl;
```
Comment (ngokevin) [inline src/core/component.js]: Could document `@this` is the component

File: src/core/component.js [hunk 5]
```diff
@@ -346,3 +351,40 @@ function extendProperties (dest, source, isSinglePropSchema) {
...
+/**
+ * Play component by adding tick behavior and calling play handler.
+ *
+ * @param component {object} - Component to play.
+ * @param sceneEl {Element} - Scene, needed to add the tick behavior.
+ */
+function wrapPlay (playMethod) {
+  return function play () {
+    var sceneEl = this.el.sceneEl;
+    var canPlay = this.el.isPlaying && !this.isPlaying;
```
Comment (ngokevin) [inline src/core/component.js]: Probably `shouldPlay`

File: tests/core/component.test.js [hunk 6]
```diff
@@ -343,4 +343,87 @@ suite('Component', function () {
       assert.equal(HTMLElement.prototype.getAttribute.call(el, 'dummy'), 'color:blue');
     });
   });
+
+  suite('play', function () {
+    setup(function () {
+      components.dummy = undefined;
+      var playStub = this.playStub = sinon.stub();
+      registerComponent('dummy', {
+        schema: {color: { default: 'red' }},
```
Comment (ngokevin) [inline tests/core/component.test.js]: don't really need to define a schema

File: src/core/component.js (hunk 2 above)
Comment (dmarcos) [inline src/core/component.js]: Can you rephrase? What's the confusing part? What the line does it's to call the play method on the component if the corresponding entity is playing.

Comment (ngokevin) [inline src/core/component.js]: The confusing part is that `isPlaying` implies that it's already playing, and therefore we wouldn't need to call `play()`.

Comment (ngokevin) [inline src/core/component.js]: Ah never mind, I confuse entity playing vs component playing.

Comment (dmarcos) [inline src/core/component.js]: `el` referes to the entity and `this` to the component

Comment (dmarcos) [inline src/core/component.js]: for an entity that is playing we have to play its components

Comment (ngokevin) [inline src/core/component.js]: Wouldn't hurt to comment anyways.

```Play the component if the entity is playing.```
